CMakePresets.json

```diff
@@ -32,7 +32,7 @@
       "description": "Inherited by all Unix-like presets for debugging. Overrides conda CXX flags for debugging",
       "hidden": true,
       "cacheVariables": {
-        "CMAKE_CXX_FLAGS_DEBUG": "-g -Og",
+        "CMAKE_CXX_FLAGS_DEBUG": "-g -O0 -fno-inline",
```

Comment:
I prefer you turn off the warning in the offending source file rather than globally. [Here is an example in napi5.cpp](https://github.com/mantidproject/mantid/blob/main/Framework/LegacyNexus/src/napi5.cpp#L172-L180)

Reply:
Oops, it's my mistake that these changes made it to the PR, I'll undo them.
I only had these for better debugging. `pdb` was unable to resolve inline functions, for instance.